vshard/storage/init.lua

```diff
@@ -3180,6 +3180,41 @@ local function storage_ref(rid, timeout)
     return bucket_count()
 end
 
+--
+-- Lookup for absent active buckets.
+--
+-- @param bucket_ids List of bucket identifiers.
+-- @return A dictionary with a list of bucket identifiers which are
+--         not present on the current instance.
+--
+local function storage_moved_buckets(bucket_ids)
+    local allstatus = consts.BUCKET
+    local moved_buckets = {}
+    for _, bucket_id in pairs(bucket_ids) do
```

Comment:
Shouldn't we check for doubled `bucket_id` in this loop?

Reply:
Not really. Duplicate filtering is already done by the router in `buckets_group()` function. This moved-buckets collection is internal API, so we don't need to support the case when it contains duplicates and can be sure the router sends the unique collection.